dashboard/src/components/AppUpgrade/AppUpgrade.tsx

```diff
@@ -39,26 +40,71 @@ interface IAppUpgradeProps {
   fetchChartVersions: (id: string) => Promise<IChartVersion[]>;
   getAppWithUpdateInfo: (releaseName: string, namespace: string) => void;
   getChartVersion: (id: string, chartVersion: string) => void;
+  getDeployedChartVersion: (id: string, chartVersion: string) => void;
   push: (location: string) => RouterAction;
   goBack: () => RouterAction;
   fetchRepositories: () => void;
-  enableBasicForm: boolean;
 }
 
 interface IAppUpgradeState {
-  selectRepoForm: JSX.Element;
+  repo?: IAppRepository;
 }
 
 class AppUpgrade extends React.Component<IAppUpgradeProps, IAppUpgradeState> {
+  public state: IAppUpgradeState = {};
+
   public componentDidMount() {
     const { releaseName, getAppWithUpdateInfo, namespace, fetchRepositories } = this.props;
     getAppWithUpdateInfo(releaseName, namespace);
     fetchRepositories();
   }
 
+  public componentDidUpdate(prevProps: IAppUpgradeProps) {
+    const { repos, app } = this.props;
+    let repo = this.state.repo;
+    // Retrieve the current repo
+    if (!repo) {
+      repo = this.props.repo;
+      if (repo && repo.metadata) {
+        // If the repository comes from the properties, use it
+        this.setState({ repo });
+      } else {
+        // If there is update info about the app we can automatically chose the repository
+        // with the latest version
+        if (app && app.updateInfo) {
+          const repoWithLatest = repos.find(
+            r => r.metadata.name === (app.updateInfo && app.updateInfo.repository.name),
+          );
+          if (repoWithLatest) {
+            this.setState({ repo: repoWithLatest });
+            repo = repoWithLatest;
+            this.setState({ repo });
```

Comment:
Looks like 2 of these lines are redundant?

Reply:
oops, yes